src/controller/event/handler/internal/artifact.go

```diff
@@ -16,19 +16,63 @@ package internal
 
 import (
 	"context"
-	"github.com/goharbor/harbor/src/lib/config"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
 	"time"
 
 	"github.com/goharbor/harbor/src/controller/artifact"
 	"github.com/goharbor/harbor/src/controller/event"
 	"github.com/goharbor/harbor/src/controller/repository"
 	"github.com/goharbor/harbor/src/controller/tag"
+	"github.com/goharbor/harbor/src/lib/config"
 	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/orm"
 	"github.com/goharbor/harbor/src/lib/q"
 )
 
+const (
+	// defaultAsyncFlushDuration is the default flush interval.
+	defaultAsyncFlushDuration = 10 * time.Second
+)
+
+var (
+	asyncFlushDuration time.Duration
+)
+
+func init() {
+	// get async flush duration from env, if not provide,
+	// use default value: 10*time.second
+	envDuration := os.Getenv("ARTIFACT_PULL_TIME_ASYNC_FLUSH_DURATION")
```

Comment:
change another name as it will handle pull count as well?

Reply:
ARTIFACT_ASYNC_FLUSH_DURATION